### ISSUE TITLE
Remove check in formatter checkbox

### ISSUE THREAD

**Issue by dpatel20** [olifolkerd/tabulator#404]
Hi -

I have a formatter set up as follows:

```JS
var checkboxIcon = function(cell, formatterParams){ //plain text value
    return '<input id="checkBox" type="checkbox">';
};
```
```JS
{title:"D", formatter:checkboxIcon, width:40, align:"center", cellClick:rowCheckboxHandler},
```
I use the checkbox to allow the user to select rows to be deleted (server side) using a `[DELETE]` button.

Whenever the checkbox is clicked the `rowCheckboxHandler` will add or remove the row to/from a 'to be deleted' list. Then when the user presses `[DELETE]`, the system will delete everything in the list on the server side.

Now my issue: I need to uncheck all the 'checked' checkboxes but I have no idea how to access the checkboxes. How can I access formatters as they are not data? 

Also, I can't use jquery to find them because tabulator uses a virtual DOM so some of the checked items might not be loaded (besides, I tried it and it at least for the visible rows and it did not work anyway!).

PS: I am happy to have another method to delete items if you have a different suggestion?

**Comment by dpatel20** [comment]
I got it to work in the following manner:

```JS
for(var i=0; i<fixtureDeleteList.length; i++){
	var index = fixtureDeleteList[i].id;
	var cb = $("#example-table-demo").tabulator("getRow", index).getCells()[0].getElement().find(">:first-child");
	cb.prop('checked', false); 
}
```
However, I have an issues with this: If I ever decide to change the order (unlikely) I have to manually  change `[0]` to reference the correct column. Since icon/button columns do not/cannot have a field then I cannot use `getCell(column)`.

**Comment by olifolkerd** [comment]
Hey,

Using a formatter in this way is very dangerous, as the data will not persist if any part of the table is redrawn, which can happen for a wide variety of reasons.

I would strongly suggest you either use:

-  the [Selectable Rows](http://olifolkerd.github.io/tabulator/docs/#selectable) functionality, which lets you select rows and get access to that selection
- an editor to store the value in the row data
- or a formatter with an icon and a ***cellClick*** event handler that then calls the ```cell.setValue()``` to set a value on the row
- or uses the ***cellClick*** callback to push the rowComponent ```cell.getRow()``` onto an array that you can then iterate through and call the ***delete*** function on the row component

Let me know if you would like a walk through of any of those options

Cheers

Oli

**Comment by dpatel20** [comment]
Hi Oli -

Darn, I just got this working with the formatter method. :(

I'm not sure which of the options you mention is best so here are some comments that might help narrow it down:

- I can't use selectable rows as the table is editable, so the user may click on a row for editing purposes.
- I liked the checkbox method as it gave the user a visual of which rows have been selected for deletion.
- Using the index ensured that only what the user selected would be deleted (I'm not using row number or anything like that, which could change).

Is the issue with my method the fact it is a formatter? I do not see how this is different to your 3rd and 4th options. In the 4th, you are using the `cellClick` option to push a row. I am essentially doing the same thing - I push the data to a delete list (I search my downloaded data for the matching index and push that to a delete list for later deletion on the server side).

I used the formatter method because I needed to add a column to the table and could not see any other method to do this without adding an extra field to my data (which I wish to avoid for editing and saving purposes).

Sorry for the long post. Hope that clarifies my thoughts/questions.
